gapis/api/vulkan/api/renderpass_framebuffer.api

```diff
@@ -163,100 +242,210 @@ cmd VkResult vkCreateRenderPass(
-          if ext.aspectReferenceCount != as!u32(0) {
-            if ext.pAspectReferences == null {
-              vkErrorNullPointer("VkInputAttachmentAspectReference(KHR)")
-            } else {
-              for j in (0 .. ext.aspectReferenceCount) {
-                // The intput attachment aspect info is only cached but not used for now, so the
-                // dependency can be wrong as a draw call may depends on just one aspect of a
-                // depth stencil image, as specified in the input attachment aspect info, while
-                // GAPID will still treat both aspects as dependees.
-                // TODO: Once the CL that enables us getting used descriptors from shader is in,
-                // we can pull input_attachment_index from the information of the used descriptors
-                // then use the input attachment aspect info to fix the dependency.
-                renderPass.InputAttachmentAspectInfo.AspectReferences[j] = ext.pAspectReferences[0]
-              }
-            }
+
+          for j in (0 .. ext.aspectReferenceCount) {
+            // The input attachment aspect info is only cached but not used for now, so the
+            // dependency can be wrong as a draw call may depends on just one aspect of a
+            // depth stencil image, as specified in the input attachment aspect info, while
+            // GAPID will still treat both aspects as dependees.
+            // TODO: Once the CL that enables us getting used descriptors from shader is in,
+            // we can pull input_attachment_index from the information of the used descriptors
+            // then use the input attachment aspect info to fix the dependency.
+            aspectReference := ext.pAspectReferences[0]
+            subPassIndex := aspectReference.subpass
+            attachmentIndex := aspectReference.inputAttachmentIndex
+
+            attachment := renderPass.SubpassDescriptions[subPassIndex].InputAttachments[attachmentIndex]
+            attachment.AspectMask = aspectReference.aspectMask
+            renderPass.SubpassDescriptions[subPassIndex].InputAttachments[attachmentIndex] = attachment
           }
         }
         case VK_STRUCTURE_TYPE_RENDER_PASS_MULTIVIEW_CREATE_INFO: {
           ext := as!VkRenderPassMultiviewCreateInfo*(next.Ptr)[0]
 
-          renderPass.MultiviewInfo = new!RenderPassMultiviewCreateInfo(
-            SubpassCount: ext.subpassCount,
-            DependencyCount: ext.dependencyCount,
-            CorrelationMaskCount: ext.correlationMaskCount,
-          )
-
           if ext.pViewMasks != null {
-            renderPass.MultiviewInfo.PViewMasks = clone(
-              as!const u32*(ext.pViewMasks)[0:ext.subpassCount])
+            viewMasks := ext.pViewMasks[0:ext.subpassCount]
+            for j in (0 .. ext.subpassCount) {
+              desc := renderPass.SubpassDescriptions[j]
+              desc.ViewMask = viewMasks[j]
+              renderPass.SubpassDescriptions[j] = desc
```

Comment:
why not `renderPass.SubpassDescrptions[j].ViewMask = viewMasks[j]`? does that not compile or fail in some other way?

Reply:
Unfortunately, it does not compile. It is ugly and unnecessary but I had to do this. Maybe there is a way to fix this from the gapil but I have no idea how.

Reply:
Thanks, I suspected as much. This is probably to make the memory read/write tracking happy.